server/app/com/xsn/explorer/modules/XSNServiceModule.scala

```diff
@@ -1,11 +1,24 @@
 package com.xsn.explorer.modules
 
-import com.google.inject.AbstractModule
-import com.xsn.explorer.services.{XSNService, XSNServiceRPCImpl}
+import akka.actor.{ActorSystem, Scheduler}
+import com.google.inject.{AbstractModule, Provides}
+import com.xsn.explorer.services.{RetryConfig, XSNService, XSNServiceRPCImpl}
+
+import scala.concurrent.duration._
 
 class XSNServiceModule extends AbstractModule {
 
   override def configure(): Unit = {
     bind(classOf[XSNService]).to(classOf[XSNServiceRPCImpl])
   }
+
+  @Provides
+  def retryConfig(): RetryConfig = {
+    RetryConfig(1.second, 65.seconds)
+  }
+
+  @Provides
+  def scheduler(actorSystem: ActorSystem): Scheduler = {
```

Comment:
This could likely go in another module, I'm not sure which one is the more adequate, let's create a new module `RetryableFutureModule` which provides the scheduler and the config.